src/lxcfs.c

```diff
@@ -157,18 +159,26 @@ static void do_reload(void)
 
 static void up_users(void)
 {
+	int nr_users;
+
+	nr_users = atomic_fetch_inc(&users_count);
+	if (!need_reload)
+		return;
+
 	users_lock();
-	if (users_count == 0 && need_reload)
+	if (nr_users == 0) {
```

Comment:
I think the `nr_users == 0` check might fail if there were ongoing requests when the first thread after `need_reload` became `true` enters this.

I think we need to decrement after the first `if` again and introduce a second condvar guarding the `do_reload()` which gets triggered in `down_user` if the count drops to 0 and `need_reload` is set?

Reply:
Hm, `nr_users` is retrieved via `atomic_fetch_inc()` so it shouldn't change anymore after we've retrieved it.